### PR TITLE
Add PyTorch version environment variable, to facilitate SMTT

### PR DESCRIPTION
**Pull request by yongyanrao**
*Issue #, if available:*

*Description of changes:* feature: add PyTorch version environment variable, to facilitate SMTT

Please refer to SMTT PR: https://github.com/aws/sagemaker-training-toolkit/pull/182

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker_pytorch_container/training.py [hunk 1]
```diff
@@ -96,6 +96,13 @@ def train(training_environment):
         capture_error = False
         logger.info(f'capture_error is {capture_error}. Default is True')
 
+    try:
+        import torch
+
+        os.environ["SM_DLC_TORCH_VERSION"] = torch.__version__
+    except Exception:
```
Comment (yl-to) [inline src/sagemaker_pytorch_container/training.py:103]: can we catch some exceptions here? like import error or module not found error.
Also indicate torch is not found in error msg.

File: src/sagemaker_pytorch_container/training.py [hunk 2]
```diff
@@ -96,6 +96,13 @@ def train(training_environment):
         capture_error = False
         logger.info(f'capture_error is {capture_error}. Default is True')
 
+    try:
```
Comment (yl-to) [inline src/sagemaker_pytorch_container/training.py:99]: can we write a function like _set_nccl_environment function above?
like _set_torch_version_environment

**Comment by satishpasumarthi** [comment]
Link SM toolkit PR in the description and viceversa for easier tracking

File: src/sagemaker_pytorch_container/training.py [hunk 3]
```diff
@@ -96,6 +96,13 @@ def train(training_environment):
...
+        import torch
+
+        os.environ["SM_DLC_TORCH_VERSION"] = torch.__version__
+    except Exception:
+        pass
```
Comment (satishpasumarthi) [inline src/sagemaker_pytorch_container/training.py:104]: is `pass` intentional ?

Comment (satishpasumarthi) [reply src/sagemaker_pytorch_container/training.py:104]: Please follow https://github.com/aws/sagemaker-pytorch-training-toolkit/blob/master/CONTRIBUTING.md#committing-your-change

**Review by satishpasumarthi — CHANGES_REQUESTED** [review]
Left comments

File: src/sagemaker_pytorch_container/training.py (hunk 3 above)
Comment (satishpasumarthi) [reply src/sagemaker_pytorch_container/training.py:104]: You should catch the exception and log it as info probably.

File: src/sagemaker_pytorch_container/training.py (hunk 2 above)
Comment (yongyanrao) [reply src/sagemaker_pytorch_container/training.py:99]: Yes.

File: src/sagemaker_pytorch_container/training.py (hunk 3 above)
Comment (yongyanrao) [reply src/sagemaker_pytorch_container/training.py]: Yes.

**Review by yl-to — APPROVED** [review]
Approving since the unit tests are passing. The error is not related to current PR changes.

**Comment by satishpasumarthi** [comment]
Ok to merge if local testing passes.

**Review by satishpasumarthi — APPROVED** [review]
LGTM

**Pull request merged**